acceptance/tests/base/dsl/helpers/host_helpers_test.rb

```diff
@@ -0,0 +1,1377 @@
+    remote_tmpdir = tmpdir_on hosts.first
+    on hosts, "mkdir -p #{remote_tmpdir}"
+    remote_filename = File.join(remote_tmpdir, "testfile.txt")
+    create_remote_file hosts, remote_filename, "contents\n"
+    hosts.each do |host|
+      remote_contents = on(host, "cat #{remote_filename}").stdout
+      assert_equal "contents\n", remote_contents
+    end
+  end
+
+  step "#create_remote_file create remote files on all remote hosts, when given an array, using scp" do
+    remote_tmpdir = tmpdir_on hosts.first
+    on hosts, "mkdir -p #{remote_tmpdir}"
+    remote_filename = File.join(remote_tmpdir, "testfile.txt")
+    create_remote_file hosts, remote_filename, "contents\n", { :protocol => 'scp' }
+    hosts.each do |host|
+      remote_contents = on(host, "cat #{remote_filename}").stdout
+      assert_equal "contents\n", remote_contents
+    end
+  end
+
+  if hosts.first.is_cygwin?
+    # NOTE: rsync methods are not working currently on windows platforms. Would
+    #       expect this to be documented better.
+
+    step "#create_remote_file create remote files on all remote hosts, when given an array, using rsync" do
+      remote_tmpdir = tmpdir_on hosts.first
+      on hosts, "mkdir -p #{remote_tmpdir}"
+      remote_filename = File.join(remote_tmpdir, "testfile.txt")
+      create_remote_file hosts, remote_filename, "contents\n", { :protocol => 'rsync' }
+      hosts.each do |host|
+        assert_raises Beaker::Host::CommandFailure do
+          remote_contents = on(host, "cat #{remote_filename}").stdout
+        end
+      end
+    end
+
+  else
+
+    step "#create_remote_file create remote files on all remote hosts, when given an array, using rsync" do
+      remote_tmpdir = tmpdir_on hosts.first
+      on hosts, "mkdir -p #{remote_tmpdir}"
+      remote_filename = File.join(remote_tmpdir, "testfile.txt")
+      create_remote_file hosts, remote_filename, "contents\n", { :protocol => 'rsync' }
+      hosts.each do |host|
+        remote_contents = on(host, "cat #{remote_filename}").stdout
+        assert_equal "contents\n", remote_contents
+      end
+    end
+  end
+
+  step "#run_script_on fails when the local script cannot be found" do
+    assert_raises IOError do
+      run_script_on hosts.first, "/non/existent/testfile.sh"
+    end
+  end
+
+  step "#run_script_on fails when there is an error running the remote script" do
+    Dir.mktmpdir do |local_dir|
+      local_filename = File.join(local_dir, "testfile.sh")
+      File.open(local_filename, "w") do |local_file|
+        local_file.puts "exit 1"
+      end
+      FileUtils.chmod "a+x", local_filename
+
+      assert_raises Beaker::Host::CommandFailure do
+        run_script_on hosts.first, local_filename
+      end
+    end
+  end
+
+  step "#run_script_on passes along options when running the remote command" do
+    Dir.mktmpdir do |local_dir|
+      local_filename = File.join(local_dir, "testfile.sh")
```

Comment:
We should probably consider what we want to do with test artifacts like this - or, at least, use temporary files and appropriate cleanup so that we don't have carry-over between steps.